cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -779,6 +787,28 @@ Result<std::vector<int>> ParquetFileFragment::FilterRowGroups(
   return row_groups;
 }
 
+Result<std::vector<int>> ParquetFileFragment::FilterRangeRowGroups(
+    int64_t start_offset, int64_t length) {
+  std::vector<int> row_groups;
+  for (int row_group : *row_groups_) {
+    auto rg_metadata = metadata_->RowGroup(row_group);
```

Comment:
`RowGroupMetadata` has a optional `file_offset()`, when `file_offset() == 0`, would that be ok?